org.uqbar.project.wollok.tests/src/org/uqbar/project/wollok/tests/interpreter/SetTestCase.xtend

```diff
@@ -0,0 +1,83 @@
+package org.uqbar.project.wollok.tests.interpreter
+
+import org.junit.Test
+
+/**
+ * @author matifreyre
+ */
+class SetTestCase extends CollectionTestCase {
+	
+	@Test
+	def void unionWithEmptySet() {
+		'''
+		program p {
+			const aSet = #{1, 2, 3}
+			
+			assert.equals(aSet, aSet.union(#{}))
+			assert.equals(aSet, #{}.union(aSet))
+		}'''.interpretPropagatingErrors
+	}
+	
+	@Test
+	def void unionWithNonEmptySets() {
+		'''
+		program p {
+			const aSet = #{1, 2, 3}
+			const anotherSet = #{3, 4, 5}
+			const unionSet = #{1, 2, 3, 4, 5}
+			
+			assert.equals(aSet, aSet.union(aSet))
+			assert.equals(unionSet, aSet.union(anotherSet))
```

Comment:
1 test = 1 assertion? I like it more too, but I see it a bit extreme
sometimes.
Knowing that the whole project already has a different approach, I rather
be consistent with the rest of the project.
Of course we can create a request to refactor the whole project tests
anytime.

On Wed, Jun 15, 2016 at 10:22 PM, Franco Leonardo Bulgarelli <
notifications@github.com> wrote:

> In
> org.uqbar.project.wollok.tests/src/org/uqbar/project/wollok/tests/interpreter/SetTestCase.xtend
> <https://github.com/uqbar-project/wollok/pull/744#discussion_r67274609>:
>
> > +			
> > +			assert.equals(aSet, aSet.union(#{}))
> > +			assert.equals(aSet, #{}.union(aSet))
> > +		}'''.interpretPropagatingErrors
> > +	}
> > +	
> > +	@Test
> > +	def void unionWithNonEmptySets() {
> > +		'''
> > +		program p {
> > +			const aSet = #{1, 2, 3}
> > +			const anotherSet = #{3, 4, 5}
> > +			const unionSet = #{1, 2, 3, 4, 5}
> > +			
> > +			assert.equals(aSet, aSet.union(aSet))
> > +			assert.equals(unionSet, aSet.union(anotherSet))
>
> Could we perhaps define separate tests for separate assertions?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/uqbar-project/wollok/pull/744/files/43910e97fe384ed85998456476fc455b092f32fc#r67274609>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AJ-EZtQdVfd9UEdU2uGi27n4r-HDlzhYks5qMKVwgaJpZM4I2zBU>
> .
>